### ISSUE TITLE
Google PubSub Extension is not working in camel-k

### ISSUE THREAD

**Issue by oscerd** [apache/camel-quarkus#2796]
I was testing a development kamelet for google pubsub source. The projectId is set in the Kamelet, but I'm getting:

```
[1] Jun 16, 2021 10:35:32 AM io.quarkus.config
[1] WARN: Unrecognized configuration key "%s" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
[1] Jun 16, 2021 10:35:32 AM io.quarkus.config
[1] WARN: Unrecognized configuration key "%s" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
[1] One or more configuration errors have prevented the application from starting. The errors are:
[1]   - SRCFG00014: The config property quarkus.google.cloud.project-id is required but it could not be found in any config source
[1]   - SRCFG00014: The config property quarkus.google.cloud.project-id is required but it could not be found in any config source
```

```yaml
  flow:
    from:
      uri: "google-pubsub:{{projectId}}:{{subscriptionName}}"
      parameters:
        .
        .
        .
        .
      steps:
      - to: "kamelet:sink"
```

The projectId is set at kamelet level and should be set a path level in the route.

**Comment by oscerd** [comment]
This is with camel-k 1.5.0-SNAPSHOT, camel-k-runtime 1.9.0-SNAPSHOT base on 2.0.0-M2

**Comment by jamesnetherton** [comment]
Relates to https://github.com/apache/camel-quarkus/issues/2858.

**Comment by jamesnetherton** [comment]
This should be fixed since CQ 2.2.0. @oscerd maybe you can confirm?

**Comment by oscerd** [comment]
Yes, it works fine with 2.2.0 :-)

Thanks for helping
